crates/testing/tests/tests_2/byzantine_tests.rs

```diff
@@ -0,0 +1,199 @@
+                    _ => Behaviour::Standard,
+                }
+            });
+
+        let mut metadata = TestDescription {
+            // allow more time to pass in CI
+            completion_task_description: CompletionTaskDescription::TimeBasedCompletionTaskBuilder(
+                                             TimeBasedCompletionTaskDescription {
+                                                 duration: Duration::from_secs(60),
+                                             },
+                                         ),
+            behaviour,
+            ..TestDescription::default()
+        };
+
+        metadata.overall_safety_properties.num_failed_views = 2;
+        metadata.num_nodes_with_stake = 5;
+        metadata.overall_safety_properties.expected_views_to_fail = HashMap::from([
+            (ViewNumber::new(7), false),
+            (ViewNumber::new(12), false)
+        ]);
+        metadata
+    },
+);
+
+cross_tests!(
+    TestName: dishonest_da,
+    Impls: [MemoryImpl, Libp2pImpl, PushCdnImpl],
+    Types: [TestTypes],
+    Versions: [MarketplaceTestVersions],
+    Ignore: false,
+    Metadata: {
+        let behaviour = Rc::new(|node_id| {
+                let dishonest_da = DishonestDa {
+                    dishonest_at_da_cert_sent_numbers: HashSet::from([2]),
+                    total_da_certs_sent_from_node: 0,
+                    total_views_add_to_cert: 4
+                };
+                match node_id {
+                    2 => Behaviour::Byzantine(Box::new(dishonest_da)),
+                    _ => Behaviour::Standard,
+                }
+            });
+
+        let mut metadata = TestDescription {
+            // allow more time to pass in CI
+            completion_task_description: CompletionTaskDescription::TimeBasedCompletionTaskBuilder(
+                                             TimeBasedCompletionTaskDescription {
+                                                 duration: Duration::from_secs(60),
+                                             },
+                                         ),
+            behaviour,
+            ..TestDescription::default()
+        };
+
+        metadata.num_nodes_with_stake = 10;
+        metadata
+    },
+);
+
+cross_tests!(
+    TestName: view_delay,
+    Impls: [MemoryImpl, Libp2pImpl, PushCdnImpl],
+    Types: [TestTypes],
+    Versions: [MarketplaceTestVersions],
+    Ignore: false,
+    Metadata: {
+        let num_nodes_with_stake = 10;
+        let behaviour = Rc::new(|node_id| {
+                let view_delay = ViewDelay {
+                    number_of_views_to_delay: 2,
+                    received_events: HashMap::new(),
+                    vote_rcv_count: HashMap::new(),
+                    num_nodes_with_stake: 10,
+                    stop_view_delay_at_view_number: 50,
+                };
+                match node_id {
+                    8 => Behaviour::Byzantine(Box::new(view_delay)),
```

Comment:
I think in this case we want to test with many nodes having this (maybe with different delays as well)